gridstatus/isone.py

```diff
@@ -227,6 +227,89 @@ def get_load_forecast(self, date, end=None, verbose=False):
 
         return df
 
+    def get_solar_forecast(self, date, end=None, verbose=False):
+        """Return solar forecast published on a specific date
+
+        Forecast is published for 7 days and generated daily by 10 am.
+        https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-solar-power-forecast
+        """
+        return (
+            self._get_solar_or_wind_forecast(
+                date,
+                end,
+                resource_type="Solar",
+                verbose=verbose,
+            )
+            .reset_index(drop=True)
+            .sort_values(["Interval Start", "Publish Time"])
+        )
+
+    def get_wind_forecast(self, date, end=None, verbose=False):
+        """Return wind forecast published on a specific date
+
+        Forecast is published for 7 days and generated daily by 10 am.
+        https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-wind-power-forecast
+        """
+        return (
+            self._get_solar_or_wind_forecast(
+                date,
+                end,
+                resource_type="Wind",
+                verbose=verbose,
+            )
+            .reset_index(drop=True)
+            .sort_values(["Interval Start", "Publish Time"])
+        )
+
+    @support_date_range(frequency="DAY_START")
+    def _get_solar_or_wind_forecast(
+        self,
+        date,
+        end=None,
+        resource_type="Wind",
+        verbose=False,
+    ):
+        """Return solar or wind forecast published on a specific date
+
+        Resource type can be "Solar" or "Wind"
+        """
+        if date == "latest":
+            date = pd.Timestamp.now(tz=self.default_timezone)
+
+        value_name = f"{resource_type.capitalize()} Forecast"
+
+        url = (
+            f"https://www.iso-ne.com/transform/csv/wphf?start={date.strftime('%Y%m%d')}"
+        )
+
+        df = _make_request(url, skiprows=[0, 1, 2, 3, 5], verbose=verbose)
+
+        df.columns = df.iloc[0]
+        df = df.drop(columns=["D", "Date"], index=[0, 1]).reset_index(drop=True)
+
+        data = df.melt(id_vars=["Hour Ending"], var_name="Date", value_name=value_name)
+
+        data["Interval Start"] = pd.to_datetime(
+            data["Date"] + " "
+            # Subtract 1 from the hour ending to get the start of the interval
+            # Make sure to 0-pad the hour
+            + (data["Hour Ending"].astype(int) - 1).astype(str).str.zfill(2) + ":00",
+            # TODO: might need to handle DST transitions
```

Comment:
hmm, i ran this

```python
>>> import gridstatus; iso=gridstatus.ISONE(); df = iso.get_solar_forecast("nov 5, 2023")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kanter/development/gridstatus/gridstatus/isone.py", line 237, in get_solar_forecast
    self._get_solar_or_wind_forecast(
  File "/Users/kanter/development/gridstatus/gridstatus/decorators.py", line 108, in wrapped_f
    df = f(**args_dict)
  File "/Users/kanter/development/gridstatus/gridstatus/isone.py", line 298, in _get_solar_or_wind_forecast
    ).dt.tz_localize(self.default_timezone)
  File "/Users/kanter/development/gridstatus/.venv/lib/python3.10/site-packages/pandas/core/accessor.py", line 94, in f
    return self._delegate_method(name, *args, **kwargs)
  File "/Users/kanter/development/gridstatus/.venv/lib/python3.10/site-packages/pandas/core/indexes/accessors.py", line 126, in _delegate_method
    result = method(*args, **kwargs)
  File "/Users/kanter/development/gridstatus/.venv/lib/python3.10/site-packages/pandas/core/indexes/datetimes.py", line 281, in tz_localize
    arr = self._data.tz_localize(tz, ambiguous, nonexistent)
  File "/Users/kanter/development/gridstatus/.venv/lib/python3.10/site-packages/pandas/core/arrays/_mixins.py", line 85, in method
    return meth(self, *args, **kwargs)
  File "/Users/kanter/development/gridstatus/.venv/lib/python3.10/site-packages/pandas/core/arrays/datetimes.py", line 987, in tz_localize
    new_dates = tzconversion.tz_localize_to_utc(
  File "pandas/_libs/tslibs/tzconversion.pyx", line 353, in pandas._libs.tslibs.tzconversion.tz_localize_to_utc
pytz.exceptions.AmbiguousTimeError: Cannot infer dst time from 2023-11-05 01:00:00, try using the 'ambiguous' argument
```

looking at the raw data, seems like ISONE doesn't handle the repeated hour in their forecast. assuming that true, my gut would be to use the same value twice when there is a repeated hour

Reply:
Okay, ISONE, _does_ handle DST in their forecast, but it's in the forecasts before the DST change.

The data from the previous day shows the "02X" hour for the duplicated time. The solution was to drop the NA values before converting times to remove times that don't exist or are duplicated.

| "C" | "ISO New England Inc. Seven Day Wind Power Forecast Report" |               |              |              |              |              |              |              |              |         | 
|-----|-------------------------------------------------------------|---------------|--------------|--------------|--------------|--------------|--------------|--------------|--------------|---------| 
| "C" | "Filename: seven_day_wind_power_forecast_20231104.csv"      |               |              |              |              |              |              |              |              |         | 
| "C" | "Report for 11/04/2023"                                     |               |              |              |              |              |              |              |              |         | 
| "C" | "Report generated 02/22/2024 11:36:54 EST"                  |               |              |              |              |              |              |              |              |         | 
| "H" | ""                                                          |               | "Day 1"      | "Day 2"      | "Day 3"      | "Day 4"      | "Day 5"      | "Day 6"      | "Day 7"      | "Day 8" | 
| "D" | "Day"                                                       |               | "Sat"        | "Sun"        | "Mon"        | "Tue"        | "Wed"        | "Thu"        | "Fri"        |         | 
| "D" | "Date"                                                      | "Hour Ending" | "11/04/2023" | "11/05/2023" | "11/06/2023" | "11/07/2023" | "11/08/2023" | "11/09/2023" | "11/10/2023" |         | 
| "D" |                                                             | 01            |              | 96           | 282          | 961          | 1017         | 264          | 186          |         | 
| "D" |                                                             | 02            |              | 89           | 149          | 984          | 1014         | 225          | 200          |         | 
| "D" |                                                             | 02X           |              | 100          |              |              |              |              |              |         | 
| "D" |                                                             | 03            |              | 98           | 120          | 987          | 1014         | 210          | 157          |         | 
| "D" |                                                             | 04            |              | 95           | 99           | 987          | 1010         | 193          | 115          |         | 
| "D" |                                                             | 05            |              | 85           | 87           | 986          | 1011         | 181          | 84           |         | 
| "D" |                                                             | 06            |              | 82           | 68           | 969          | 1009         | 164          | 75           |         |